alluxio-ui/master/src/containers/pages/Workers/Workers.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+import {AxiosResponse} from 'axios';
+import React from 'react';
+import {connect} from 'react-redux';
+import {Alert, Progress, Table} from 'reactstrap';
+import {Dispatch} from 'redux';
+
+import {INodeInfo} from '../../../constants';
+import {IApplicationState} from '../../../store';
+import {fetchRequest} from '../../../store/workers/actions';
+import {IWorkers} from '../../../store/workers/types';
+
+interface IPropsFromState {
+  errors?: AxiosResponse;
+  loading: boolean;
+  refresh: boolean;
+  workers: IWorkers;
+}
+
+interface IPropsFromDispatch {
+  fetchRequest: typeof fetchRequest;
+}
+
+type AllProps = IPropsFromState & IPropsFromDispatch;
+
+class Workers extends React.Component<AllProps> {
+  public componentDidUpdate(prevProps: AllProps) {
+    if (this.props.refresh !== prevProps.refresh) {
+      this.props.fetchRequest();
+    }
+  }
+
+  public componentWillMount() {
+    this.props.fetchRequest();
+  }
+
+  public render() {
+    const {errors, workers} = this.props;
+
+    if (errors) {
+      return (
+        <Alert color="danger">
+          Unable to reach the api endpoint for this page.
+        </Alert>
+      );
+    }
+
+    return (
+      <div className="workers-page">
+        <div className="container-fluid">
+          <div className="row">
+            <div className="col-12">
+              <h5>Live Workers</h5>
+              <Table hover={true}>
+                <thead>
+                <tr>
+                  <th>Node Name</th>
+                  <th>[D]Worker Id</th>
+                  <th>[D]Uptime</th>
```

Comment:
These should only be displayed if debug is enabled.

Reply:
Good point, I need a new api to support this.